lib/prediction/predictor.ts

```diff
@@ -282,7 +303,8 @@ export default class Predictor {
     }
 
     private _startWorker() {
-        const worker = new Worker(`${this.id}/${this._nextId++}`, this._modeldir);
+        const inference_name = this.id.replace(/\W/g, '');
```

Comment:
Do we need some escaping mechanism instead of dropping characters? To some extent, model IDs can be user-generated, and I worry that malicious users can craft model IDs that give them access to models they do not own.

Reply:
It was unfortunate kfserving service name can only be alphanumeric.  This name is also hard coded in ingress routing rule.
I'm open to ideas. We can add a config map of model name to inference service name.  But maintaining it would be painful.   I suppose we will have  access control checks before the prediction? We can also force model names to be alphanumeric only.

Reply:
Do they allow any special character? -, _, anything? If they do, we can do quoted-printable escaping and replacing all special characters with `_` followed by the hex representation of the character.
I tried looking up the docs but didn't find anything definitive, although I did find examples with _ and -

Reply:
Kubenetes name allows '- and .  but I did not have success using them.  The inference service will not create resources after I applied.  I didn't see any error message but I haven't had the time find the root cause. I suspect one of resources failed to create due to the name.

Reply:
We can also use 0 as special character. So any non-alphanumeric can be represented by 0??  (where ?? is the ASCII hex code).  If 0 exists in the name, we can escape it using 00?

Reply:
Yeah 0 as special character would work.

Reply:
cool.. will do that in another PR.